src/main/java/com/microsoft/azure/AzureVMAgent.java

```diff
@@ -307,10 +306,13 @@ public void setCleanUpAction(CleanUpAction cleanUpAction, Localizable cleanUpRea
         }
         if (this.toComputer()!= null) {
             AzureVMComputer computer = (AzureVMComputer)this.toComputer();
-            // Set the machine temporarily offline machine with an offline reason.
-            computer.setTemporarilyOffline(true, OfflineCause.create(cleanUpReason));
-            // Reset the "by user" bit.
-            computer.setSetOfflineByUser(false);
+            if(computer != null) //findbugs was complaining about not checking here. Just doing it to please the gods
```

Comment:
This can be simplified by moving the scope of `AzureVMComputer computer` outside the conditional on line 307.

That way, instead of calling `this.toComputer()` twice, it could be called once, and then the nested null check wouldn't need to exist since 307 would already cover that